lib/echosign/credentials.rb

```diff
@@ -1,31 +1,113 @@
 module Echosign
-  class Credentials < Hash
+  class Credentials
+
+    require 'oauth2'
 
     include Validatable
 
-    # Builds a Credentials object
-    #
-    # @param app_id [String] Application key
-    # @param app_secret [String] Application secret
-    # @param api_key [String] API Key
-    # @param email [String] User email
-    # @param password [String] User password
-    # @return [Echosign::Credentials] Echosign credentials
-    def initialize(app_id, app_secret, api_key, email, password)
-      merge!(
-        {
-          applicationCredentials:  {
-            applicationId: app_id,
-            applicationSecret: app_secret
-          },
-          userCredentials: {
-            apiKey: api_key,
-            email: email,
-            password: password 
-          }
-        }
-      )
+    OAUTH_SITE = 'https://secure.echosign.com'
+    AUTHORIZE_PATH = '/public/oauth'
+    TOKEN_PATH = '/oauth/token'
+    REFRESH_PATH = '/oauth/refresh'
+    REVOKE_PATH = '/oauth/revoke'
+
+    attr_reader :access_token, :refresh_token, :expires_at
+
+    # Builds an Credentials object
+    #
+    # @param client_id [String] Client ID
+    # @param client_secret [String] Client secret
+    #
+    # @return [Echosign::Credentials] Echosign OAuth2 wrapper object
+    def initialize(client_id, client_secret)
```

Comment:
Allow `redirect_uri` as an optional param here, and remove it as a param in the other methods.